File: Front/EventsPro-App/src/app/models/speaker.ts
import { BaseEntity } from "./base-entity";
import { SocialNetwork } from "./social-network";
import { Event } from "./event";

export interface Speaker extends BaseEntity {
    name: string;
    description: string;
    imageURL: string;
    phone: string;
    email: string;
    socialNetworks: SocialNetwork[];
    speakersEvents: Event[];
}

File: Back/src/EventsPro.Domain/Entities/Speaker.cs
using EventsPro.Domain.Entities.Base;

namespace EventsPro.Domain.Entities
{
    public class Speaker : BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string ImageURL { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public IEnumerable<SocialNetwork>? SocialNetworks { get; set; }
        public IEnumerable<SpeakerEvent>? SpeakersEvents { get; set; }
    }
}